src/db/connection.cr

```diff
@@ -21,10 +21,12 @@ module DB
   abstract class Connection
     include Disposable
     include SessionMethods(Connection, Statement)
+    include BeginTransaction
 
     # :nodoc:
     getter database
     @statements_cache = StringKeyCache(Statement).new
+    @transaction = false
```

Comment:
Well, tracking the object force either a refactor of how the BEGIN/CREATE SAVEPOINT is done OR a :nodoc: setter is needed. Because we need to flag the connection as in-transaction **before** the sql command is sent.

I think I would prefer to keep bools for now, [the required change](https://gist.github.com/bcardiff/bd4fbfed985439880d60a9f1332e4d2b) is not that nice. Either way it should not impact to drivers implementors since they don't need to inherit Transaction class, or if done, they should call base constructor.